src/index.js

```diff
@@ -1,78 +1,359 @@
-        cb()
-      })
+  /**
+   * Register the rendezvous protocol in the libp2p node.
+   * @returns {void}
+   */
+  start () {
+    if (this._registrarId) {
+      return
+    }
+
+    log('starting')
+
+    // Create and start Rendezvous server if enabled
+    if (this._serverOptions.enabled) {
+      this._server = new Server(this._libp2p, this._serverOptions)
+      this._server.start()
+    }
+
+    // register protocol with topology
+    const topology = new MulticodecTopology({
+      multicodecs: PROTOCOL_MULTICODEC,
+      handlers: {
+        onConnect: this._onPeerConnected,
+        onDisconnect: this._onPeerDisconnected
+      }
     })
+    this._registrarId = this._registrar.register(topology)
+
+    log('started')
+  }
+
+  /**
+   * Unregister the rendezvous protocol and the streams with other peers will be closed.
+   * @returns {void}
+   */
+  stop () {
+    if (!this._registrarId) {
+      return
+    }
+
+    log('stopping')
+
+    clearInterval(this._interval)
+
+    // unregister protocol and handlers
+    this._registrar.unregister(this._registrarId)
+    if (this._serverOptions.enabled) {
+      this._server.stop()
+    }
+
+    this._registrarId = undefined
+    this._rendezvousPoints.clear()
+    this._cookiesSelf.clear()
+
+    log('stopped')
   }
 
-  _rpc (cmd, ...a) { // TODO: add. round-robin / multicast / anycast?
-    this.peers[0][cmd](...a)
+  /**
+   * Registrar notifies a connection successfully with rendezvous protocol.
+   * @private
+   * @param {PeerId} peerId remote peer-id
+   * @param {Connection} conn connection to the peer
+   */
+  _onPeerConnected (peerId, conn) {
+    const idB58Str = peerId.toB58String()
+    log('connected', idB58Str)
+
+    this._rendezvousPoints.set(idB58Str, { connection: conn })
   }
 
-  register (ns, peer, cb) {
-    this._rpc('register', ns, peer, 0, cb) // TODO: interface does not expose ttl option?!
+  /**
+   * Registrar notifies a closing connection with rendezvous protocol.
+   * @private
+   * @param {PeerId} peerId peerId
+   */
+  _onPeerDisconnected (peerId) {
+    const idB58Str = peerId.toB58String()
+    log('disconnected', idB58Str)
+
+    this._rendezvousPoints.delete(idB58Str)
+
+    if (this._server) {
+      this._server.removePeerRegistrations(peerId)
+    }
   }
 
-  discover (ns, limit, cookie, cb) {
-    if (typeof cookie === 'function') {
-      cb = cookie
-      cookie = Buffer.from('')
+  /**
+   * Register the peer in a given namespace
+   * @param {string} ns
+   * @param {object} [options]
+   * @param {number} [options.ttl = 7200e3] registration ttl in ms (minimum 120)
+   * @returns {Promise<number>} rendezvous register ttl.
+   */
+  async register (ns, { ttl = 7200e3 } = {}) {
+    if (!ns) {
+      throw errCode(new Error('a namespace must be provided'), errCodes.INVALID_NAMESPACE)
     }
-    if (typeof limit === 'function') {
-      cookie = Buffer.from('')
-      cb = limit
-      limit = 0
+
+    if (ttl < 120) {
```

Comment:
120ms?! I assume this should be 2 minutes.

It looks like the spec doesn't have any minimum bound here mentioned, we should probably add that as a recommendation.

Reply:
I grabbed this from the go PR. But, considering https://github.com/libp2p/js-libp2p-rendezvous/pull/6#discussion_r438074063 I think it makes sense for this to live in the server and not here. We should just have a friendly default